pyramus/src/main/java/fi/otavanopisto/pyramus/webhooks/WebhookEventListeners.java

```diff
@@ -0,0 +1,229 @@
+  
+  /* Course Staff Member */
+  
+  public void onCourseStaffMemberCreated(@Observes(during=TransactionPhase.AFTER_SUCCESS) CourseStaffMemberCreatedEvent event) {
+    webhooks.sendWebhookNotification(new WebhookCourseStaffMemberCreatePayload(event.getCourseStaffMemberId(), event.getCourseId(), event.getStaffMemberId()));
+  }
+  
+  public synchronized void onCourseStaffMemberUpdatedBeforeCompletion(@Observes(during=TransactionPhase.BEFORE_COMPLETION) CourseStaffMemberUpdatedEvent event) {
+    datas.addUpdatedCourseStaffMember(event.getCourseStaffMemberId(), event.getCourseId(), event.getStaffMemberId());
+  }
+
+  public synchronized void onCourseStaffMemberUpdatedAfterSuccess(@Observes(during=TransactionPhase.AFTER_SUCCESS) CourseStaffMemberUpdatedEvent event) {
+    List<Long> updatedCourseStaffMemberIds = datas.retrieveUpdatedCourseStaffMemberIds();
+    
+    for (Long updatedCourseStaffMemberId : updatedCourseStaffMemberIds) {
+      Long courseId = datas.getCourseStaffMemberCourseId(updatedCourseStaffMemberId);
+      Long staffMemberId = datas.getCourseStaffMemberStaffMemberId(updatedCourseStaffMemberId);
+      webhooks.sendWebhookNotification(new WebhookCourseStaffMemberUpdatePayload(updatedCourseStaffMemberId, courseId, staffMemberId));
+    }
+    
+    datas.clearUpdatedCourseStaffMemberIds();
+  }
+  
+  public void onCourseStaffMemberDeleted(@Observes(during=TransactionPhase.AFTER_SUCCESS) CourseStaffMemberDeletedEvent event) {
+    webhooks.sendWebhookNotification(new WebhookCourseStaffMemberDeletePayload(event.getCourseStaffMemberId(), event.getCourseId(), event.getStaffMemberId()));
+  }
+  
+  /* Course Student */
+  
+  public void onCourseStudentCreated(@Observes(during=TransactionPhase.AFTER_SUCCESS) CourseStudentCreatedEvent event) {
+    webhooks.sendWebhookNotification(new WebhookCourseStudentCreatePayload(event.getCourseStudentId(), event.getCourseId(), event.getStudentId()));
+  }
+  
+  public synchronized void onCourseStudentUpdatedBeforeCompletion(@Observes(during=TransactionPhase.BEFORE_COMPLETION) CourseStudentUpdatedEvent event) {
+    CourseStudent courseStudent = DAOFactory.getInstance().getCourseStudentDAO().findById(event.getCourseStudentId());
+    datas.addUpdatedCourseStudent(event.getCourseStudentId(), event.getCourseId(), courseStudent.getStudent().getId());
+  }
+  
+  public synchronized void onCourseStudentUpdatedAfterSuccess(@Observes(during=TransactionPhase.AFTER_SUCCESS) CourseStudentUpdatedEvent event) {
+    List<Long> updatedCourseStudentIds = datas.retrieveUpdatedCourseStudentIds();
+    
+    for (Long updatedCourseStudentId : updatedCourseStudentIds) {
+      Long courseId = datas.getCourseStudentCourseId(updatedCourseStudentId);
+      Long studentId = datas.getCourseStudentStudentId(updatedCourseStudentId);
+      webhooks.sendWebhookNotification(new WebhookCourseStudentUpdatePayload(updatedCourseStudentId, courseId, studentId));
+    }
+    
+    datas.clearUpdatedCourseStudentIds();
```

Comment:
See above comment.

Reply:
This is old code as-is. There is a separate issue where these should be covered: https://github.com/otavanopisto/pyramus/issues/536.